client-src/elements/chromedash-feature-table.ts

```diff
@@ -93,14 +93,7 @@ export class ChromedashFeatureTable extends LitElement {
     window.csClient
       .getPendingGates()
       .then(res => {
-        const gatesByFID: Record<number, GateDict[]> = {};
-        for (const g of res.gates) {
-          if (!gatesByFID.hasOwnProperty(g.feature_id)) {
-            gatesByFID[g.feature_id] = [];
-          }
-          gatesByFID[g.feature_id].push(g);
-        }
-        this.gates = gatesByFID;
+        this.gates = Object.groupBy(res.gates, ({feature_id}) => feature_id);
```

Comment:
This one is a clear upgrade.  Jules also detected this one.

Personally, I have mixed feelings about using destructuring in the arrow function parameters.  So, there's basically two changes happening even in this small part of the PR: a library usage upgrade, and a coding style change.